django/views/generic/base.py

```diff
@@ -71,12 +70,14 @@ def view(request, *args, **kwargs):
         view.view_class = cls
         view.view_initkwargs = initkwargs
 
-        # take name and docstring from class
-        update_wrapper(view, cls, updated=())
+        # Note that __name__ and __qualname__ are intentionally left
+        # unchanged as view_class should be used to robustly determine
+        # the name of the view instead.
+        view.__doc__ = cls.__doc__
+        view.__module__ = cls.__module__
+        view.__annotations__ = cls.dispatch.__annotations__
+        view.__dict__.update(cls.dispatch.__dict__)
```

Comment:
I think I'd have left in a comment like about decorators though, since it's not obvious why we are doing this line.
```suggestion
        # Copy possible attributes set by decorators, e.g. @csrf_exempt, from
        # the dispatch method.
        view.__dict__.update(cls.dispatch.__dict__)
```

Reply:
My bad, I got a little confused.

Reply:
No worries - probably a poor description on my part in the earlier comment :)